### PR TITLE
Update Armeria and Netty dependencies

### PR DESCRIPTION
**Pull request by minwoox**
Armeria 0.99.8 -> 0.99.9
Netty 4.1.50.Final -> 4.1.51.Final

### CODE REVIEW

File: zipkin-server/src/main/java/zipkin2/server/internal/ZipkinHttpCollector.java [hunk 1]
```diff
@@ -168,7 +159,7 @@ HttpResponse validateAndStoreSpans(SpanBytesDecoder decoder, ServiceRequestConte
           return null;
         }
       } finally {
-        ReferenceCountUtil.release(content);
+        PooledObjects.close(content);
```
Comment (anuraaga) [inline zipkin-server/src/main/java/zipkin2/server/internal/ZipkinHttpCollector.java]: Don't we have try / resources now?

**Review by anuraaga — APPROVED** [review]

File: zipkin-server/src/main/java/zipkin2/server/internal/ZipkinHttpCollector.java (hunk 1 above)
Comment (minwoox) [reply zipkin-server/src/main/java/zipkin2/server/internal/ZipkinHttpCollector.java]: Ah let me use that. Thanks!

**Review by codefromthecrypt — APPROVED** [review]
Amazing initiative. Thanks!

File: zipkin-server/src/test/java/zipkin2/server/internal/health/ITZipkinHealth.java [hunk 2]
```diff
@@ -73,7 +73,7 @@
 
     // ensure we don't track health in prometheus
     assertThat(scrape())
-      .doesNotContain("health");
+      .doesNotContain("health_check");
```
Comment (minwoox) [inline zipkin-server/src/test/java/zipkin2/server/internal/health/ITZipkinHealth.java]: Is there any better string that only exposed by prometheus?
We have changed the meter tags to contain method and service name.
For example `scrape()` contains `method=getHealth` and `service=server.internal.health.ITzipkinHealth` which made this test failed. 😓

File: zipkin-server/src/test/java/zipkin2/server/internal/prometheus/ITZipkinMetrics.java [hunk 3]
```diff
@@ -77,7 +77,7 @@
 
     // ensure we don't track prometheus, UI requests in prometheus
     assertThat(scrape())
-      .doesNotContain("prometheus")
+      .doesNotContain("prometheus_notifications")
```
Comment (minwoox) [inline zipkin-server/src/test/java/zipkin2/server/internal/prometheus/ITZipkinMetrics.java]: This one as well. Any better suggestion please?

File: zipkin-server/src/test/java/zipkin2/server/internal/health/ITZipkinHealth.java (hunk 2 above)
Comment (codefromthecrypt) [reply zipkin-server/src/test/java/zipkin2/server/internal/health/ITZipkinHealth.java]: but isn't this possibly a correct positive? why would we be adding metrics for the scrape endpoint? Maybe you can dump the actual contents here.

File: zipkin-server/src/test/java/zipkin2/server/internal/prometheus/ITZipkinMetrics.java (hunk 3 above)
Comment (codefromthecrypt) [reply zipkin-server/src/test/java/zipkin2/server/internal/prometheus/ITZipkinMetrics.java]: similarly we intentionally shouldn't see any meta stats about prometheus. Possibly this is a correct glitch and something else we need to disable now. Stats cost money (especially in SaaS) and we don't want to blanket add things especially monitoring the monitor without being conscious about it.

**Review by codefromthecrypt — COMMENTED** [review]
Thanks for the help.. wondering where the new metrics are coming from, if there's a PR. Maybe it can help us hunt how to disable them.

File: zipkin-server/src/test/java/zipkin2/server/internal/prometheus/ITZipkinMetrics.java (hunk 3 above)
Comment (minwoox) [reply zipkin-server/src/test/java/zipkin2/server/internal/prometheus/ITZipkinMetrics.java]: The metric contains `service="zipkin2.server.internal.prometheus.ZipkinMetricsController"` so it's the problem. 😄 

Here's the story.
We fixed to add FQCN of the service name for HTTP services as a metric tag in 0.99.7. https://github.com/line/armeria/pull/2780
So if the call is made to `ZipkinMetricsController.fetchMetricsFromMicrometer()`, 
one of the metric recorded is `...method="fetchMetricsFromMicrometer",service="zipkin2.server.internal.prometheus.ZipkinMetricsController",quantile="0.0"...`

But we had a bug in 0.99.8 that the `MetricCollectingService` is not automatically added.
If Armeria server is created using spring boot integration and `ArmeriaSettings.isEnableMetric()` returns `true`, `MetricCollectingService` is added by default.
https://github.com/line/armeria/pull/2898/files#diff-58a90bb70d1f88b9c2f45ee9bddd4468R126
https://github.com/line/armeria/pull/2898/files#diff-58a90bb70d1f88b9c2f45ee9bddd4468R148

So from 0.99.9 we add `MetricCollectingService` the `Server` so all the requests and responses are recorded by default.

Now, `metricsIsOK()` is executed before `prometheusIsOK()` is executed during the test which records the metric about `service="zipkin2.server.internal.prometheus.ZipkinMetricsController"` which contains `prometheus` that make the test fails.

Comment (minwoox) [reply zipkin-server/src/test/java/zipkin2/server/internal/prometheus/ITZipkinMetrics.java]: So it's better not to add `MetricCollectingService` by default using `ArmeriaSettings`? 🤔

**Comment by codefromthecrypt** [comment]
ok I may not fully understand but it seems at least dirty stats could be at
play here. that and possibly a controller we have should be marked opt out
of metrics or both..

there's an expensive annotation for spring boot tests called
DirtiesApplicationContext or something for the former..

**Comment by minwoox** [comment]
> that and possibly a controller we have should be marked opt out
of metrics or both..

Or we could just remove the `MetricCollectingService` that is added by Armeria spring boot autoconfiguration. It seems like Zipkin is collecting metrics of the requests under `/zipkin/api` and `/api`, we might not need the default `MetricCollectingService`. 😆 
https://github.com/openzipkin/zipkin/blob/master/zipkin-server/src/main/java/zipkin2/server/internal/prometheus/ZipkinPrometheusMetricsConfiguration.java#L91-L96

Let me just disable the `ArmeriaSettings.enableMetrics`.

**Comment by codefromthecrypt** [comment]
Thanks @minwoox! I think maybe only main metrics we need to make sure work are the elasticsearch client ones particularly as we are using client-side loadbalancing. I'm not sure we don't already have IT for that.

**Comment by minwoox** [comment]
> I think maybe only main metrics we need to make sure work are the elasticsearch client ones particularly as we are using client-side loadbalancing. I'm not sure we don't already have IT for that.

I counddn't find the IT for that. But `ArmeriaSettings.enableMetrics` is just about server-side, so no effect for the clients. 😉 

And the test failure seems flaky, do you have a chance to re-run it, please?

**Comment by codefromthecrypt** [comment]
Excellent. Thanks for the help and followup @minwoox Thanks for the review feedback @anuraaga !

**Comment by minwoox** [comment]
Thanks for reviewing! I hope I can have more chances to do more work on Zipkin. 😄

**Pull request merged**